### DISCUSSION TITLE
anvil --optimism functionality

### DISCUSSION THREAD

**Discussion by HrikB** [foundry-rs/foundry#8264] [Q&A] [answered]
What exactly does the optimism flag do? I haven't noticed a change in the tx receipt structure or code at the predeploys.

**Comment by urosognjenovic** [comment]
The Foundry docs only have this: 

```
--optimism
          Run an Optimism chain
          
          [aliases: optimism]
```

**Comment by zerosnacks** [comment, marked as answer]
The `--optimism` flag exists to enable Optimism specific functionality which includes transaction types like `DepositTransaction` and enables the setting of hardforks specific to Optimism.
